aspnetcore/fundamentals/use-http-context.md

```diff
@@ -0,0 +1,277 @@
+
+For information on how to read content from `BodyReader`, see [I/O pipelines PipeReader](/dotnet/standard/io/pipelines#pipereader).
+
+## `HttpResponse`
+
+<xref:Microsoft.AspNetCore.Http.HttpContext.Response?displayProperty=nameWithType> provides access to <xref:Microsoft.AspNetCore.Http.HttpResponse>. `HttpResponse` is used to set information on the HTTP response sent back to the client.
+
+Commonly used properties on `HttpResponse` include:
+
+|Property|Description|Example|
+|--|--|--|--|
+|<xref:Microsoft.AspNetCore.Http.HttpResponse.StatusCode?displayProperty=nameWithType>|The response code. Must be set before writing to the response body.|`200`|
+|<xref:Microsoft.AspNetCore.Http.HttpResponse.ContentType?displayProperty=nameWithType>|The response `content-type` header. Must be set before writing to the response body.|`application/json`|
+|<xref:Microsoft.AspNetCore.Http.HttpResponse.Headers?displayProperty=nameWithType>|A collection of response headers. Must be set before writing to the response body.|`server=Kestrel`<br />`x-custom-header=MyValue`|
+|<xref:Microsoft.AspNetCore.Http.HttpResponse.Body?displayProperty=nameWithType>|A <xref:System.IO.Stream> for writing the response body.|
+
+### Set response headers
+
+<xref:Microsoft.AspNetCore.Http.HttpResponse.Headers?displayProperty=nameWithType> provides access to the response headers sent with the HTTP response. There are two ways to access headers using this collection:
+
+* Provide the header name to the indexer on the header collection. The header name isn't case-sensitive. The indexer can access any header value.
+* The header collection also has properties for getting and setting commonly used HTTP headers. The properties provide a fast, strongly typed way to access headers.
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", (HttpResponse response) =>
+{
+    response.Headers.CacheControl = "no-cache";
+    response.Headers["x-custom-header"] = "Custom value";
+
+    return Results.File(File.OpenRead("helloworld.txt"));
+});
+
+app.Run();
+```
+
+An app can't modify headers after the response has started, which sends the headers to the client. A response is started by flushing the response body or calling <xref:Microsoft.AspNetCore.Http.HttpResponse.StartAsync(System.Threading.CancellationToken)?displayProperty=nameWithType>. An error is thrown when attempting to modify headers after the response has started.
+
+### Set response trailers
+
+HTTP/2 and HTTP/3 support response trailers. Trailers are headers sent with the response after the response body is complete. Because trailers are sent after the response body, trailers can be added to the response at any time.
+
+Set trailers using `AppendTrailer`:
```

Comment:
I couldn't figure out how to create a working xref link to AppendTrailer.

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.http.responsetrailerextensions.appendtrailer?view=aspnetcore-6.0

Please help doc experts 🙏

Reply:
Figured it out.